<launch>

 	<arg name="blob_name" default="output" />
    <arg name="visualize" default="true"/>
   <!-- launch video stream -->
   <include file="$(find video_stream_opencv)/launch/camera.launch" >
        <!-- node name and ros graph name -->
        <arg name="camera_name" value="videofile" />
        <!-- full path to the video file -->
        <!-- wget http://techslides.com/demos/sample-videos/small.mp4 -O /tmp/small.mp4 -->
       <arg name="video_stream_provider" value="/home/tsa/TSA/track_ws/src/track_and_chase/files/robotics_problem.mp4" />
        <!-- set camera fps to (video files not affected) -->
        <!-- <arg name="set_camera_fps" value="30"/> -->
        <!-- set buffer queue size of frame capturing to -->
        <arg name="buffer_queue_size" value="1000" />
        <!-- throttling the querying of frames to -->
        <arg name="fps" value="30" />
        <!-- setting frame_id -->
        <arg name="frame_id" value="videofile_frame" />
        <!-- camera info loading, take care as it needs the "file:///" at the start , e.g.:
        "file:///$(find your_camera_package)/config/your_camera.yaml" -->
        <arg name="camera_info_url" value="" />
        <!-- flip the image horizontally (mirror it) -->
        <arg name="flip_horizontal" value="false" />
        <!-- flip the image vertically -->
        <arg name="flip_vertical" value="false" />
        <!-- enable looping playback -->
        <arg name="loop_videofile" value="true" />
        <!-- start frame of video -->
        <arg name="start_frame" default="0"/>
        <!-- stop frame of video, -1 means the end of video -->
        <arg name="stop_frame" default="-1"/>
        <!-- visualize on an image_view window the stream generated -->
        <arg name="visualize" value="true" />
   </include>
 <node pkg="track_and_chase" type="track_and_chase_node" name="track_and_chase_node" output="screen">
      <param name="loop_rate" value="30" />
 </node> 

 <node pkg="track_and_chase" type="chase_robot_node" name="chase_robot_node" output="screen"> 
      <param name="loop_rate" value="30" />
      <param name="max_linear_accel_" value="300" />
      <param name="max_angular_accel_" value="100" />
</node>



<group ns="$(arg blob_name)">
    <node if="$(arg visualize)" name="$(arg blob_name)_image_view" pkg="image_view" type="image_view">
	    	<remap from="image" to="image_raw" />
    </node>
</group>


</launch>